src/node/historical_queries.h

```diff
@@ -64,12 +106,44 @@ namespace ccf::historical
         requests.erase(old_idx);
       }
 
-      // Try to insert new request
-      const auto ib = requests.insert(std::make_pair(idx, Request{}));
+      Request request;
+      request.current_stage = RequestStage::Fetching;
+      consensus::Index first_idx_to_fetch = idx;
+
+      // If the target historical entry cannot be deserialised with the first
+      // known ledger secret, record the target idx and fetch the previous
+      // historical ledger secret.
+      auto first_known_ledger_secret = get_first_known_ledger_secret();
```

Comment:
`first_known_ledger_secret` is already a long variable name, and then we actually use `.first` and `.second` from it - I have no idea what those are. Can we destructure this here, something like:
```
auto [first_secret_idx, first_secret] = get_first_known_ledger_secret();
```